rust-bins/wallet-notes/README.md

```diff
@@ -444,6 +444,57 @@ The return value is a a JSON array with JSON objects as entries. Each object has
 
 With meaning that can be discerned from their names.
 
+## get_identity_keys_and_randomness
+
+Semantics: Deterministically derives id_cred_sec, prf_key and blinding randomness for an identity.
+
+This function takes as input a NUL-terminated UTF8-encoded string. The string
+must be a valid JSON object with fields
+
+- `"seed"` ... the seed used to derive keys from, as a hex string.
+
+- `"net"` ... determines whether to derive keys for Mainnet or a testnet. Has to be "Mainnet" or "Testnet", all other values will fail.
+
+- `"identityIndex"` ... the index of the identity to derive keys and randomness for, a u32 value
+
+The returned value is a JSON object with the following fields:
+
+- `"idCredSec"` ... the id_cred_sec as a hex encoded string.
+
+- `"prfKey"` ... the prf_key as a hex encoded string.
+
+- `"blindingRandomness"` ... the blinding randomness as a hex encoded string.
+
+An example input to this request is in the file [get_identity_keys_and_randomness-input.json](files/get_identity_keys_and_randomness-input.json).
+An example output to this request is in the file [get_identity_keys_and_randomness-output.json](files/get_identity_keys_and_randomness-output.json).
+
+## get_account_keys_and_randomness
+
+Semantics: Deterministically derives signing key, verification key and attribute randomness for an account.
+
+This function takes as input a NUL-terminated UTF8-encoded string. The string
+must be a valid JSON object with fields
+
+- `"seed"` ... the seed used to derive keys from, as a hex string.
+
+- `"net"` ... determines whether to derive keys for Mainnet or a testnet. Has to be "Mainnet" or "Testnet", all other values will fail.
+
+- `"identityIndex"` ... the index of the identity to derive keys and randomness for, a u32 value
+
+- `"accountCredentialIndex"` ... the index of the account credential to derive keys and randomness for, a u32 value
+
+The returned value is a JSON object with the following fields:
+
+- `"signingKey"` ... the account signing key as a hex encoded string.
+
+- `"verifyKey"` ... the account verification key as a hex encoded string.
```

Comment:
Well... I was thinking that they probably got both the private and public key generated by our libraries currently, but I didn't actually check. My assumption was also that they just got `IdCredSec` generated and that our libraries automatically calculates `IdCredPub` from this when needed, but I didn't check this either. We should verify exactly what keys they store today so we can give them the same from this library.

I may not be able to find out today.

Reply:
Currently they have two calls
- the first `create_id_request_and_private_data` which returns both idCredSec and idCredPub and everything else (e.g., randomness for retrieving the signature)
- `create_credential` which returns the credential to send to the chain, as well as verifyKey and signKey (as well as commitments randomness for all the commitments that were made to the attributes in the credential.

Reply:
But then shouldn't `get_identity_keys_and_randomness` here also return `idCredPub` to be consistent?

Reply:
It's a bit complicated there, because `idCredPub = g^idCredSec` where `g` is the first part of the on-chain commitment key.
So that'd have to be passed into this function as well.

As I wrote above (see also #160 ) we need to rework a little bit the wallet API anyhow. I think for this function returning just idCredSec is OK.